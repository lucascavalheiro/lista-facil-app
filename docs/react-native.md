# Getting Started with React Native

## Setting Up Your Machine for App Development

- iOS (You require a Mac to develop iOS Apps) Simply download XCode
- [Android on Mac](https://medium.com/pvtl/react-native-android-development-on-mac-ef7481f65e47#.w2hel88zy)
- Android on Windows - coming soon...

## Install React Native

[React Native - Get Started Guide](https://facebook.github.io/react-native/docs/getting-started.html) 

## Learn React

[React Native Express](http://www.reactnativeexpress.com/) is a great site to get you started, specifically:

- [Get your head around ES6](http://www.reactnativeexpress.com/es6)
- [What is JSX?](http://www.reactnativeexpress.com/jsx)
- [What are Components?](http://www.reactnativeexpress.com/components)
- [React State](http://www.reactnativeexpress.com/data_component_state)
- [Redux](http://www.reactnativeexpress.com/redux)

Once you've got your head around the basics, checkout the [React Native](https://facebook.github.io/react-native/) website, specifically:

- Go through ['The Basics'](https://facebook.github.io/react-native/docs/props.html)
- Gain an understanding of the [components](https://facebook.github.io/react-native/docs/activityindicator.html) React Native provides out of the box
