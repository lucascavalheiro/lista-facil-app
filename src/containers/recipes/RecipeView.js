/**
 * Recipe View Screen
 *  - The individual recipe screen
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Image,
  ScrollView,
  StyleSheet,
} from 'react-native';

// Consts and Libs
import { AppStyles, AppSizes } from '@theme/';

// Components
import { Card, Spacer, Text } from '@ui/';

/* Styles ==================================================================== */
const styles = StyleSheet.create({
  featuredImage: {
    left: 0,
    right: 0,
    height: AppSizes.screen.height * 0.2,
    resizeMode: 'cover',
  },
});

/* Component ==================================================================== */
class RecipeView extends Component {
  static componentName = 'RecipeView';

  static propTypes = {
    recipe: PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      body: PropTypes.string.isRequired,
      image: PropTypes.string,
      ingredients: PropTypes.arrayOf(PropTypes.string),
      method: PropTypes.arrayOf(PropTypes.string),
    }).isRequired,
  }

  /**
    * Ingredients
    */
  renderIngredients = (ingredients) => {
    const jsx = [];
    let iterator = 1;

    ingredients.forEach((item) => {
      jsx.push(
        <View key={`ingredient-${iterator}`} style={[AppStyles.row]}>
          <View><Text> - </Text></View>
          <View style={[AppStyles.paddingLeftSml, AppStyles.flex1]}>
            <Text>{item.toString()}</Text>
          </View>
        </View>,
      );
      iterator += 1;
    });

    return jsx;
  }

  /**
    * Method
    */
  renderMethod = (method) => {
    const jsx = [];
    let iterator = 1;

    method.forEach((item) => {
      jsx.push(
        <View key={`method-${iterator}`} style={[AppStyles.row]}>
          <View><Text> {iterator}. </Text></View>
          <View style={[AppStyles.paddingBottomSml, AppStyles.paddingLeftSml, AppStyles.flex1]}>
            <Text>{item.toString()}</Text>
          </View>
        </View>,
      );
      iterator += 1;
    });

    return jsx;
  }

  render = () => {
    const { title, body, image, ingredients, method } = this.props.recipe;

    return (
      <ScrollView style={[AppStyles.container]}>
        {image !== '' &&
          <Image
            source={{ uri: image }}
            style={[styles.featuredImage]}
          />
        }

        <Card>
          <Text h2>{title.rendered}</Text>
          <Text>{body}</Text>
        </Card>

        {ingredients ?
          <Card>
            <Text h2>Ingredients</Text>
            {this.renderIngredients(ingredients)}
          </Card>
        : null}

        {method ?
          <Card>
            <Text h2>Method</Text>
            {this.renderMethod(method)}
          </Card>
        : null}

        <Spacer size={20} />
      </ScrollView>
    );
  }
}

/* Export Component ==================================================================== */
export default RecipeView;
