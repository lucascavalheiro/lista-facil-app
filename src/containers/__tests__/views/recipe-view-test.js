/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import RecipeView from '@containers/recipes/RecipeView';

it('RecipeView renders correctly', () => {
  const thisRecipe = {
    id: 1,
    title: 'Recipe Title',
    body: 'Blah di blah lorem ipsum',
    image: 'http://placehold.it/300x100',
    ingredients: ['Hello world', 'Hello world', 'Hello world'],
    method: ['And again', 'And again', 'And again'],
  };

  const tree = renderer.create(
    <RecipeView recipe={thisRecipe} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
