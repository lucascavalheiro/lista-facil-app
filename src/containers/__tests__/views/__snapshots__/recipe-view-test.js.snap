// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeView renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
    ]
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "http://placehold.it/300x100",
        }
      }
      style={
        Array [
          Object {
            "height": 266.8,
            "left": 0,
            "resizeMode": "cover",
            "right": 0,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#D0D1D5",
              "borderRadius": 2,
            },
            Array [],
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          h1={false}
          h2={true}
          h3={false}
          h4={false}
          h5={false}
          onPress={null}
          p={false}
          style={
            Array [
              Object {
                "color": "#0E4EF8",
                "fontFamily": "HelveticaNeue",
                "fontSize": 21,
                "fontWeight": "800",
                "left": 0,
                "lineHeight": 26,
                "margin": 0,
                "marginBottom": 4,
                "right": 0,
              },
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          onPress={null}
          p={false}
          style={
            Array [
              Object {
                "color": "#222222",
                "fontFamily": "HelveticaNeue",
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 18,
              },
            ]
          }
        >
          Blah di blah lorem ipsum
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#D0D1D5",
              "borderRadius": 2,
            },
            Array [],
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          h1={false}
          h2={true}
          h3={false}
          h4={false}
          h5={false}
          onPress={null}
          p={false}
          style={
            Array [
              Object {
                "color": "#0E4EF8",
                "fontFamily": "HelveticaNeue",
                "fontSize": 21,
                "fontWeight": "800",
                "left": 0,
                "lineHeight": 26,
                "margin": 0,
                "marginBottom": 4,
                "right": 0,
              },
            ]
          }
        >
          Ingredients
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 0,
                "right": 0,
              },
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
               - 
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingLeft": 10,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
              Hello world
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 0,
                "right": 0,
              },
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
               - 
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingLeft": 10,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
              Hello world
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 0,
                "right": 0,
              },
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
               - 
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingLeft": 10,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
              Hello world
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#D0D1D5",
              "borderRadius": 2,
            },
            Array [],
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          h1={false}
          h2={true}
          h3={false}
          h4={false}
          h5={false}
          onPress={null}
          p={false}
          style={
            Array [
              Object {
                "color": "#0E4EF8",
                "fontFamily": "HelveticaNeue",
                "fontSize": 21,
                "fontWeight": "800",
                "left": 0,
                "lineHeight": 26,
                "margin": 0,
                "marginBottom": 4,
                "right": 0,
              },
            ]
          }
        >
          Method
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 0,
                "right": 0,
              },
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
               
              1
              . 
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                },
                Object {
                  "paddingLeft": 10,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
              And again
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 0,
                "right": 0,
              },
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
               
              2
              . 
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                },
                Object {
                  "paddingLeft": 10,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
              And again
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 0,
                "right": 0,
              },
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
               
              3
              . 
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                },
                Object {
                  "paddingLeft": 10,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            >
              And again
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 1,
          "left": 0,
          "marginTop": 19,
          "right": 0,
        }
      }
    />
  </View>
</RCTScrollView>
`;
